### DISCUSSION TITLE
Error `element.getAncestors is not a function`

### DISCUSSION THREAD

**Discussion by pandika-csgi** [clientIO/joint#2314] [Q&A] [answered]
### Introduction

Hi, I don't know why this suddenly happened
it was fine before, but after updating one of the packages in `package.json` I got this issue
I'm using Angular appIt

### Steps to reproduce

1. Change 1 package inside `package.json`
2. Delete `node_modules` and `package-lock.json`
3. `npm install`
4. Reload page and when click an element, issue occurred

![image](https://github.com/clientIO/joint/assets/106044262/6d53e700-0ae3-4a6b-9730-d67d96095ae2)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
What application specifically do you use? Angular KitchenSink? What version of `JointJS+` do you have?

**Reply by pandika-csgi** [reply]
Yes, I am using Angular KitchenSink, I'm using Angular 13, JointJS from rappid.tgz is JointJS v3.7.5 (2023-08-02) - JavaScript diagramming library

**Reply by kumilingus** [reply]
Could you share the `package.json` of yours with us?

The steps to reproduce are not useful here, since we don't have your Angular 13 application (we do not ship Angular 13 application).

When you click on an element, the element is added to the collection. The elements in the collection are sorted by the depth (`element.getAncestors().length`). Each cell (element and link) have the method defined. In your case, an object that is not `dia.Cell` is added to the collection. You might try to debug this and find out what the object is.

Make sure you are using the up-to-date dependencies.
```json
  "dependencies": {
    "backbone": "~1.4.1",
    "dagre": "~0.8.5",
    "graphlib": "~2.1.8",
    "jquery": "~3.6.4",
    "lodash": "~4.17.21"
  },
  ```

**Reply by pandika-csgi** [reply]
sorry I was using JointJS v3.7.4 from my `rappid.tgz` archive but I dont know why JointJS package updated to v3.7.5 after run `npm install` again, I opened rappid.tgz archive there a tilde "jointjs":"~3.7.4" inside `package.json`, is this the root cause of problem?

this is my package.json
```json
"dependencies": {
    "@angular/core": "^13.3.2",
    "@clientio/rappid": "file:rappid.tgz",
    "@mdi/font": "^6.5.95",
    "@popperjs/core": "^2.10.2",
    "backbone": "^1.4.1",
    "bootstrap": "^5.1.3",
    "dagre": "0.8.4",
    "jquery-nestable": "^0.8.0",
    "lodash": "^4.17.21",
    "rxjs": "6.5.4",
    "tslib": "^2.0.0",
    "zone.js": "~0.11.4"
  }
```

**Reply by kumilingus** [reply, marked as answer]
I think the problem is that you are using 2 different versions of `Backbone`.
Backbone `v1.5.0` has been released recently. Your dependencies allows installing the minor versions (`^1.4.1`), but JointJS dependency is restricted to patch versions only.
You should remove `Backbone` from your `package.json` or change it to `~1.4.1`. Alternatively edit the JointJS+ package.json and change `Backbone` to `1.5.0` (didn't it test yet, but assume compatibility).
